stdlib/public/runtime/Heap.cpp

```diff
@@ -32,10 +32,11 @@ using namespace swift;
 /// On Apple platforms, \c malloc() is always 16-byte aligned.
 static constexpr size_t MALLOC_ALIGN_MASK = 15;
 
-#elif defined(__linux__) || defined(_WIN32)
+#elif defined(__linux__) || defined(_WIN32) || defined(__wasi__)
```

Comment:
Is this also the case for wasm64? I see wasi-libc has some `#if defined(__wasm64__)` checks in a few places.